File: 401-binary-watch/binary-watch.py
class Solution:
    def readBinaryWatch(self, turnedOn: int) -> List[str]:
        ans = []

        for hour in range(12):
            for mint in range(60):
                if (bin(hour).count("1") + bin(mint).count("1")) == turnedOn:
                    time = f"{hour}:{mint:02}"
                    ans.append(time)

        return ans

File: 401-binary-watch/binary-watch.cs
public class Solution {
    public IList<string> ReadBinaryWatch(int turnedOn) {
        List<string> ans = new List<string>();

        for (int hh = 0; hh <= 11; hh++) {
            for (int mm = 0; mm <= 59; mm++) {
                if (BitCount(hh) + BitCount(mm) == turnedOn) {
                    string hour = hh.ToString();
                    string mint = (mm < 10 ? "0" : "") + mm;
                    ans.Add(hour + ":" + mint);
                }

            }
        }

        return ans;
    }

    private int BitCount(int n) {
        return Convert.ToString(n, 2).Count(c => c == '1');
    }
}